docs/readmes/bazel/bazel.md

```diff
@@ -0,0 +1,510 @@
+- To build all targets inside of a folder, e.g. `lte/gateway`, use `bazel build //lte/gateway/...`.
+- Build one target `bazel build //lte/gateway/python/magma/mobilityd:mobilityd`, here the `//` indicate the start of the path to the target and the `:` indicated the start of the target name.
+    - The same target can be build with `bazel build //lte/gateway/python/magma/mobilityd` as this expands to the target with the same name as the folder in the absence of `...` and `:`.
+- To specify multiple targets or multiple folders simply add them to the command line, e.g. `bazel build //orc8r/gateway/... //lte/gateway/python/magma/mobilityd:mobilityd`, this example will build the mobilityd service target, as well as all targets inside the `orc8r/gateway` folder.
+
+A complete list of [command line arguments](https://bazel.build/reference/command-line-reference) can be found in the official bazel documentation.
+
+### Configurations
+
+Targets can be build with different configurations. Configurations are specified on the command line by adding the command line flag `--config=<config_name>`.  Definitions of configurations can be found in the `.bazelrc` file. A configuration is specified by a bazel verb followed by a colon, the name of the configuration and the content, e.g. `build:example --announce_rc --color=yes`.
+
+ The most important configurations for Magma are:
+
+- Plain, i.e. no special configuration, this is the development version of the targets.
+- `production`, which is the production version of the targets used for releases.
+- `asan`, a configuration with the address sanitizer enabled, used for finding memory corruption bugs.
+- `lsan`, a configuration with the leak sanitizer enabled, used for finding run-time memory leaks.
+
+### Dependencies
+
+To add a dependency to a bazel target the `deps` attribute should be used. The `deps` attribute exists for all library, binary and test rules. Dependencies are specified as a list of strings. There are several distinct types of dependencies:
+
+- [local dependencies](#local-dependencies)
+- [external dependencies](#external-dependencies)
+- [pip dependencies](#pip-dependencies-python).
+
+A simple example that illustrates these is as follows:
+
+```starlark
+py_library(
+  name = "my_lib",
+  srcs = ["my_lib.py"],
+  deps = [
+  	":my_package_dependency",
+  	"//orc8r/magma/common:my_util",
+  	"@aioh2_repo//:aioh2",
+  	requirement("grpcio"),
+  ],
+)
+```
+
+#### Local dependencies
+
+Dependencies on targets in the **same package**, i.e. the same `BUILD.bazel` file, are specified by the target name, prefixed by `:`. For example as `deps = [":my_package_dependency"]`.
+
+Dependencies on targets in **other packages within Magma** are additionally prefixed by `//` and the relative path of the respective package. For example as `deps = ["//orc8r/magma/common:my_util"]`.
```

Comment:
Isn't this the absolute path?

Reply:
I would say it is relative to the base of the repository, e.g. `docs/readmes/bazel/agw_with_bazel.md` vs `/workspaces/magma/docs/readmes/bazel/agw_with_bazel.md`. Will rephrase.